scipy/sparse/tests/test_base.py

```diff
@@ -780,7 +780,7 @@ def check_setdiag(a, b, k):
             # correct length, and too short or too long vectors
             for r in [-1, len(np.diag(a, k)), 2, 30]:
                 if r < 0:
-                    v = int(np.random.randint(1, 20, size=1))
+                    v = np.random.randint(1, 20, size=1)[0]
```

Comment:
If we are touching this already could you please use `np.random.choice(range(1, 20))`  which a bit more readable than the mysterious [0]?

Reply:
Did that.